charts/nexus3/templates/ingress.yaml

```
@@ -37,8 +37,9 @@ spec:
               servicePort: http
             {{- end }}
   {{- end }}
-  {{- range .Values.service.additionalPorts }}
-    - host: {{ .host | quote }}
+  {{- range additionalPort := .Values.service.additionalPorts }}
```

```suggestion
  {{- $range additionalPort := .Values.service.additionalPorts }}
```

I think this should fix the failing CI.

---

No because `range` is Helm function, and return this error:

```
Error: parse error at (nexus3/templates/ingress.yaml:40): undefined variable "$range"
```

Also I found a bug with `ternary` because the last value isn't a boolean.

Works with this configuration:

```
diff --git i/charts/nexus3/templates/ingress.yaml w/charts/nexus3/templates/ingress.yaml
index 68336f6..2fc6c9d 100644
--- i/charts/nexus3/templates/ingress.yaml
+++ w/charts/nexus3/templates/ingress.yaml
@@ -37,8 +37,8 @@ spec:
               servicePort: http
             {{- end }}
   {{- end }}
-  {{- range additionalPort := .Values.service.additionalPorts }}
-  {{- range ternary $additionalPort.hosts (list $additionalPort.host) $additionalPort.hosts }}
+  {{- range $additionalPort := .Values.service.additionalPorts }}
+  {{- range ternary (list $additionalPort.host) $additionalPort.hosts (empty $additionalPort.hosts) }}
     - host: {{ . | quote }}
       http:
         paths:
```

Tested with this values (host & hosts):

```
  additionalPorts:
    - port: 8082
      name: docker-group
      containerPort: 8082
      host: nexus-docker.local
    - port: 8083
      name: docker-hosted
      containerPort: 8083
      hosts:
        - nexus-docker-hosted-1.local
        - nexus-docker-hosted-2.local
```

Helm template response:

```
$ helm template -s templates/ingress.yaml -f values.yaml nexus3 . | yq -r '.spec.rules[].host'
nexus.local
nexus-docker.local
nexus-docker-hosted-1.local
nexus-docker-hosted-2.local
```

Good for you?

---

Sorry @djerfy I typoed where the `$` needed adding but it looks like you figured it out. Could you push the changes, squash your commits and rebase?